presto-main/src/main/java/com/facebook/presto/server/ResourceGroupStateInfo.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server;
+
+import com.facebook.presto.spi.resourceGroups.ResourceGroupId;
+import com.facebook.presto.spi.resourceGroups.ResourceGroupState;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import io.airlift.units.DataSize;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class ResourceGroupStateInfo
+{
+    private final ResourceGroupId id;
+    private final ResourceGroupState state;
+
+    private final DataSize softMemoryLimit;
+    private final DataSize memoryUsage;
+
+    private final int maxRunningQueries;
+    private final List<QueryStateInfo> runningQueries;
+
+    private final int maxQueuedQueries;
+    private final int numQueuedQueries;
+
+    @JsonCreator
+    public ResourceGroupStateInfo(
+            @JsonProperty("id") ResourceGroupId id,
+            @JsonProperty("state") ResourceGroupState state,
+            @JsonProperty("softMemoryLimit") DataSize softMemoryLimit,
+            @JsonProperty("memoryUsage") DataSize memoryUsage,
+            @JsonProperty("maxRunningQueries") int maxRunningQueries,
+            @JsonProperty("runningQueries") List<QueryStateInfo> runningQueries,
+            @JsonProperty("maxQueuedQueries") int maxQueuedQueries,
+            @JsonProperty("numQueuedQueries") int numQueuedQueries)
```

Comment:
If you add the timeouts, I would add put all the limits in consecutive lines (maxRunning, maxQueued, softMemoryLimit, queuedTimeout, runningTimeout) to make it consistent. Otherwise this organization makes sense.